core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -220,6 +225,38 @@ private[deploy] class Worker(
     metricsSystem.getServletHandlers.foreach(webUi.attachHandler)
   }
 
+  // TODO if we're starting up multi workers under the same host, discovery script won't work.
```

Comment:
How would the `resourceFile` work under your approach on the scenario of multiple workers launched on the same host? We need to avoid multiple workers access the same external resource address.

Reply:
For the initial version, I have a base assumption here that user should be responsible for configuring the right `resourceFile` among multi workers on the same hosts.

Reply:
Yeah, technically we can do, but I think it may be more troublesome than `resourceFile` to do it. Document is good, have done in ade97c2.